breadbox/breadbox/api/temporary.py

```diff
@@ -0,0 +1,81 @@
+
+from fastapi import APIRouter, Body, Depends
+
+from breadbox.api.dependencies import get_db_with_user
+from breadbox.config import Settings, get_settings
+from breadbox.crud import dataset as dataset_crud
+from breadbox.crud import slice as slice_crud
+from breadbox.db.session import SessionWithUser
+from breadbox.schemas.custom_http_exception import UserError
+from breadbox.schemas.context import (
+    ContextRequest,
+    ContextSummary,
+    ContextMatchResponse,
+)
+
+# This temporary prefix is intended to convey to API users that these contracts may change.
+# Most of these endpoints are intended to support feature-specific functionality
+router = APIRouter(prefix="/temporary", tags=["temporary"])
+
+
+@router.post(
+    "/context",
+    operation_id="evaluate_context",
+    response_model=ContextMatchResponse,
+    response_model_exclude_none=False,
+)
+def evaluate_context(
+    db: Annotated[SessionWithUser, Depends(get_db_with_user)],
+    settings: Annotated[Settings, Depends(get_settings)],
+    context_request: Annotated[
+        ContextRequest, Body(description="A Data Explorer 2 context expression")
+    ],
+):
+    """
+    Get the full list of IDs and labels (in any dataset) which match the given context.
+    Requests may be made in either the old or new format. 
+    """
+    # Evaluate each of the dimension's given_ids against the context
+    matching_ids, matching_labels = slice_crud.get_ids_and_labels_matching_context(
+        db, settings.filestore_location, context_request.context
+    )
+
+    return ContextMatchResponse(ids=matching_ids, labels=matching_labels,)
+
+
+@router.post(
+    "/context/summary",
+    operation_id="get_context_summary",
+    response_model=ContextSummary,
+    response_model_exclude_none=False,
+)
+def get_context_summary(
+    db: Annotated[SessionWithUser, Depends(get_db_with_user)],
+    settings: Annotated[Settings, Depends(get_settings)],
+    context_request: Annotated[
+        ContextRequest, Body(description="A Data Explorer 2 context expression")
+    ],
+):
+    """
+    Get the number of matching labels and candidate labels.
+    "Candidate" labels are all labels belonging to the context's dimension type.
+    Requests may be made in either the old or new format. 
+    """
+    context = context_request.context
+    dimension_type = (
+        context.dimension_type if context.dimension_type else context.context_type
+    )
+    if dimension_type is None:
+        raise UserError(("Context requests must specify a dimension type."))
+    all_labels_by_id = dataset_crud.get_dimension_labels_by_id(db, dimension_type)
+
+    # Evaluate each of the dimension's given_ids against the context
+    ids_matching_context, _ = slice_crud.get_ids_and_labels_matching_context(
+        db, settings.filestore_location, context
+    )
+
+    return ContextSummary(
+        num_candidates=len(all_labels_by_id.keys()),
+        num_matches=len(ids_matching_context),
+    )
```

Comment:
These endpoints look very similar and I would advocate for maybe keeping the former while doing the simple counts on the frontend

Reply:
This morning, @rcreasi and I talked about this and it seems like he's on-board with combining them as long as it's still possible to get the number of "candidate" entities. I would propose that I:
1. get rid of this "summary" endpoint altogether
2. add an additional field `"num_candidates"` to the response from the other context endpoint (since it's not easy to get this information through any other breadbox endpoints) 

Does that sound reasonable to you, @jessica-cheng ?

Reply:
Sounds good!